src/null-safety/migrating.md

```diff
@@ -0,0 +1,397 @@
+flutter run, flutter test flag order might be slightly different.
+pub run --enable-experiment=non-nullable test
+aka dart test --enable-experiment=non-nullable]_**
+
+## Prepare to migrate
+
+We strongly recommend migrating code in order, 
+with the leaves of the dependency graph being migrated first.
+For example, if package C depends on package B, which depends on package A,
+then A should be migrated to null safety first, then B, then C.
+
+![Illustration of C/B/A sentence](/null-safety/migrating/null-safety-migration-order.png){:width="454px"}<br>
+
+Although you _can_ migrate before your dependencies support null safety,
+you might have to change your code when your dependencies migrate.
+For example, if you predict that a function will take a nullable parameter but
+the package migrates it to be non-nullable,
+then passing a nullable argument becomes a compile error.
+
+
+### Switch to a beta release
+
+Start the migration process by switching to the **latest beta release**
+of either the Dart SDK or the Flutter SDK:
+
+* If you use the Dart SDK, download a beta release
+  from the [Dart SDK archive][].
+* If you use the Flutter SDK,
+  switch to the beta channel:
+
+  ```terminal
+$ flutter channel beta
+$ flutter upgrade
+```
+
+[Dart SDK archive]: /tools/sdk/archive#beta-channel
+
+{{ site.alert.warn }}
+  When you return to working on production code,
+  **switch back to a stable release**
+  (for example, by running `flutter channel stable`).
+{{ site.alert.end }}
+
+
+### Wait for dependencies
+
+Get the migration state of your package's dependencies
+by using the following command:
+
+```terminal
+$ dart pub outdated --mode=null-safety
+```
+
+If the output says **all the packages fully support null safety**,
+then you can start migrating.
+
+Otherwise, check whether you can update to
+package versions that support null safety.
+Many packages have null-safe prerelease versions.
+For example, while the [pedantic package][] stable release was 1.9.2,
+it also had prereleases 1.10.0-nullsafety, 1.10.0-nullsafety.1, and so on.
+
+[pedantic package]: {{site.pub-pkg}}/pedantic
+
+If you update your pubspec, remember to upgrade before
+checking for outdated packages:
+
+```terminal
+$ dart pub upgrade
+$ dart pub outdated --mode=null-safety
+```
+
+**_[PENDING: Offer more help here. What should the SDK constraints be?
+Show updating the pubspec? Is there anything else they need to do/change?]_**
+
+
+## Migrate
+
+Most of the changes that your code needs to be null safe
+are easily predictable.
+For example, if a variable can be `null`,
+its type needs a `?` suffix.
+A named parameter that shouldn't be nullable
+needs to be marked `required`.
+
+The migration tool can make most of these changes,
+and you can add hints to customize the changes.
+Alternatively, you can
+[migrate your code by hand](#migrating-by-hand).
+
+
+### Using the migration tool
+
+**_[PENDING: Minimum sdk version in pubspec?
+dart --version requirements?]_**
```

Comment:
The `dart version` requirement is a 2.12 **beta** release.

Reply:
Is there no pubspec requirement?